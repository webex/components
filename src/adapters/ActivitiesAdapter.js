import {throwError} from 'rxjs';

import WebexAdapter from './WebexAdapter';

/**
 * This is a base class that defines the interface that maps activity data.
 * Developers that want to extend `ActivitiesAdapter` must implement all of its methods,
 * adhering to the exact parameters and structure of the returned objects.
 */
export default class ActivitiesAdapter extends WebexAdapter {
  /**
   * An activity a person performs in Webex.
   *
   * @typedef  {Object} Activity
   * @property {string}  ID             The activity identifier.
   * @property {string}  roomID         ID of the room where the activity happens.
   * @property {string}  text           Any text the activity may contain.
   * @property {string}  personID       ID of the person performing the activity.
   * @property {Date}    created        Timestamp of the time when the activity happened.
   * @property {Boolean} displayAuthor  Whether to display author information or not.
   */

  /**
   * Returns an observable that emits activity data of the given ID.
   *
   * @param {string} ID  ID of the activity to get.
   * @returns {Observable.<Activity>}
   * @memberof ActivityAdapter
   */
  // eslint-disable-next-line no-unused-vars
  getActivity(ID) {
    return throwError(new Error('getActivity(ID) must be defined in ActivitiesAdapter'));
  }
}
