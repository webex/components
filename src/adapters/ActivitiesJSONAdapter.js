import {ActivitiesAdapter} from '@webex/component-adapter-interfaces';
import {Observable} from 'rxjs';

const EMPTY_ACTION = {
  actionID: null,
  personID: null,
  roomID: null,
  type: 'submit',
  activityID: null,
  inputs: null,
  created: null,
};

// TODO: Figure out how to import JS Doc definitions and remove duplication.
/**
 * An activity a person performs in Webex.
 *
 * @external Activity
 * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/ActivitiesAdapter.js#L6}
 */

/**
 * @typedef ActivitiesJSON
 * @param {object} datasource An object that contains activities keyed by ID
 * @example
 * {
 *   "activity-1": {
 *     "ID": "activity-1",
 *     "roomID": "roomID",
 *     "text": "Hey Barbara, how is it going?",
 *     "personID": "user-2",
 *     "created": "created",
 *     "displayAuthor": false
 *   },
 *   "activity-2": {
 *     "ID": "activity-2",
 *     "roomID": "roomID",
 *     "text": "Oh no! That's terrible ðŸ˜”",
 *     "personID": "user-2",
 *     "created": "created",
 *     "displayAuthor": true
 *   }
 * }
 */

/**
 * `ActivitiesJSONAdapter` is an implementation of the `ActivitiesAdapter` interface.
 * This implementation utilizes a JSON object as its source of activity data.
 *
 * @see {@link ActivitiesJSON}
 * @implements {ActivitiesAdapter}
 */
export default class ActivitiesJSONAdapter extends ActivitiesAdapter {
  /**
   * Returns an observable that emits activity data of the given ID.
   * For this implementation, once the data is emitted, the observable completes.
   *
   * @param {string} ID ID of activity to get
   * @returns {Observable.<Activity>} Observable that emits data of the given ID
   */
  getActivity(ID) {
    return Observable.create((observer) => {
      const activity = this.datasource[ID];

      if (activity) {
        const card = activity.attachments && activity.attachments[0] && activity.attachments[0].contentType === 'application/vnd.microsoft.card.adaptive' ? activity.attachments[0].content : undefined;

        observer.next({
          ...this.datasource[ID],
          card,
        });
      } else {
        observer.error(new Error(`Could not find activity with ID "${ID}"`));
      }

      observer.complete();
    });
  }

  /**
   * Posts an attachment action, returns an observable that emits the created action
   *
   * @param {string} activityID  ID of the activity corresponding to this submit action
   * @param {object} inputs  The message content
   * @returns {Observable.<object>} Observable stream that emits data of the newly created action
   */
  postAction(activityID, inputs) {
    return Observable.create((observer) => {
      const activity = this.datasource[activityID];

      if (activity) {
        const actionID = `action-${activityID}-${activity?.actions.length + 1}`;
        const personID = 'user1';
        const newAction = {
          ...EMPTY_ACTION,
          actionID,
          personID,
          roomID: activity.roomID,
          activityID,
          inputs,
          created: new Date().toISOString(),
        };
        const activityActions = this.datasource[activityID].actions;

        activityActions.push(newAction);

        observer.next(newAction);
        observer.complete();
      } else {
        observer.error(new Error(`Unable to create an attachment action for activity with id "${activityID}"`));
      }
    });
  }
}
