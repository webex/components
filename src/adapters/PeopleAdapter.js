import {throwError} from 'rxjs';

import WebexAdapter from './WebexAdapter';

/**
 * Enum for person status values.
 *
 * @readonly
 * @enum {string}
 */
export const PersonStatus = {
  ACTIVE: 'active',
  BOT: 'bot',
  CALL: 'call',
  DO_NOT_DISTURB: 'dnd',
  INACTIVE: 'inactive',
  MEETING: 'meeting',
  OUT_OF_OFFICE: 'ooo',
  PRESENTING: 'presenting',
  SELF: 'self',
  TYPING: 'typing',
};

/**
 * This is a base class that defines the interface that maps people data.
 * Developers that want to extend `PeopleAdapter` must implement all of its methods,
 * adhering to the exact parameters and structure of the returned objects.
 */
export default class PeopleAdapter extends WebexAdapter {
  /**
   * A Person object with details about the person.
   * @typedef {Object} Person
   * @property {string}               ID          The person identifier.
   * @property {Array.<string>}       emails      An array of emails for the person.
   * @property {string}               displayName The name to be displayed for the person.
   * @property {string}               firstName   The first name of the person.
   * @property {string}               lastName    The last name of the person.
   * @property {string}               nickName    The short name for the person.
   * @property {string}               avatar      The full url to the person's avatar.
   * @property {string}               orgID       The ID of the organization the person belongs to.
   * @property {null|PersonStatus}    status      The presence status of the user. @see PersonStatus enum
   */

  /**
   * Returns an observable that emits person data of the given ID.
   *
   * @param {string} ID ID of person to get.
   * @returns {Observable.<Person>}
   * @memberof PeopleAdapter
   */
  // eslint-disable-next-line no-unused-vars
  getPerson(ID) {
    return throwError(new Error('getPerson(ID) must be defined in PeopleAdapter'));
  }
}
