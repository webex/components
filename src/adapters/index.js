export {default as WebexAdapter} from './WebexAdapter';
