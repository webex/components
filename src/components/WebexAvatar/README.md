# Webex Avatar Component

Webex avatar component displays the avatar and status of a _Webex_ user.

<p align="center">
  <img src="./WebexAvatar.png" alt="Default Webex Avatar" />
</p>

## Preview

To see all the different possible states of the Webex Avatar component,
you can view our [storybook](https://webex.github.io/components/storybook/?path=/story/webex-avatar--default)
or run the following **NPM** command:

```shell
  npm start
```

## Embed

1. Create a component adapter from which the data will be retrieved (See [adapters](../../adapters)). For instance:

    ```js
    const jsonAdapter = new WebexJSONAdapter(jsonData);
    ```

2. Create a component instance by passing the person ID as a string and
enclose it within [a data provider](../WebexDataProvider/WebexDataProvider.js)
that takes the [component data adapter](../../adapters/WebexJSONAdapter.js) that we created previously

    ```js
    <WebexDataProvider adapter={jsonAdapter}>
      <WebexAvatar personID="personID" />
    </WebexDataProvider>
    ```

The component knows how to manage its data. If anything changes in the data source that the adapter manages, the component will also update on its own.
