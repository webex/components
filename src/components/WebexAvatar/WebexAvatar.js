import React from 'react';
import PropTypes from 'prop-types';
import {Avatar} from '@momentum-ui/react';

import usePerson from '../hooks/usePerson';

export default function WebexAvatar(props) {
  let component = null;
  const person = usePerson(props.personID, props.adapter);

  if (person) {
    const {avatar, displayName, status} = person;

    component = <Avatar src={avatar} title={displayName} type={status} alt={displayName} />;
  }

  return component;
}

WebexAvatar.propTypes = {
  personID: PropTypes.string.isRequired,
  adapter: PropTypes.object.isRequired,
};
