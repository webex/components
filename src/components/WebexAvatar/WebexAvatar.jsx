import React from 'react';
import PropTypes from 'prop-types';
import {Avatar} from '@momentum-ui/react';

import {usePerson} from '../hooks';

/**
 * Displays the avatar of a Webex user.
 *
 * @param {object} props  Data passed to the component
 * @param {string} props.personID  ID of the person for which to display avatar
 * @param {boolean} props.displayStatus  Whether or not to display the user's status
 *
 * @returns {object} JSX of the component
 */
export default function WebexAvatar({personID, displayStatus}) {
  const {avatar, displayName, status} = usePerson(personID);
  const type = displayStatus ? status : null;

  return <Avatar src={avatar} title={displayName} type={type} alt={displayName} />;
}

WebexAvatar.propTypes = {
  personID: PropTypes.string.isRequired,
  displayStatus: PropTypes.bool,
};

WebexAvatar.defaultProps = {
  displayStatus: true,
};
