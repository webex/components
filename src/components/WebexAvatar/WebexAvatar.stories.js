import React from 'react';
import {PersonStatus} from '@webex/component-adapter-interfaces';
import people from '../../data/people.json';
import WebexAvatar from './WebexAvatar';

export default {
  title: 'Platform/Webex Avatar',
  component: WebexAvatar,
};

const Template = (args) => <WebexAvatar {...args} />;

export const NoStatus = Template.bind({});
NoStatus.args = {
  personID: 'default',
  displayStatus: false,
};

export const Active = Template.bind({});
Active.args = {
  personID: 'default',
};
Active.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.ACTIVE,
      },
    },
  },
};

export const Call = Template.bind({});
Call.args = {
  personID: 'default',
};
Call.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.CALL,
      },
    },
  },
};

export const Bot = Template.bind({});
Bot.args = {
  personID: 'default',
};
Bot.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.BOT,
      },
    },
  },
};

export const DoNotDisturb = Template.bind({});
DoNotDisturb.args = {
  personID: 'default',
};
DoNotDisturb.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.DO_NOT_DISTURB,
      },
    },
  },
};

export const Inactive = Template.bind({});
Inactive.args = {
  personID: 'default',
};
Inactive.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.INACTIVE,
      },
    },
  },
};

export const Meeting = Template.bind({});
Meeting.args = {
  personID: 'default',
};
Meeting.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.MEETING,
      },
    },
  },
};

export const OutOfOffice = Template.bind({});
OutOfOffice.args = {
  personID: 'default',
};
OutOfOffice.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.OUT_OF_OFFICE,
      },
    },
  },
};

export const Presenting = Template.bind({});
Presenting.args = {
  personID: 'default',
};
Presenting.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.PRESENTING,
      },
    },
  },
};

export const Self = Template.bind({});
Self.args = {
  personID: 'default',
};
Self.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.SELF,
      },
    },
  },
};

export const Typing = Template.bind({});
Typing.args = {
  personID: 'default',
};
Typing.parameters = {
  jsonData: {
    people: {
      default: {
        ...people.default,
        status: PersonStatus.TYPING,
      },
    },
  },
};

export const Invalid = Template.bind({});
Invalid.args = {
  personID: '123',
};
