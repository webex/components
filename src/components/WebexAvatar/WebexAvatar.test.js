import React from 'react';

import {PersonStatus} from '../../adapters/PeopleAdapter';

import WebexAvatar from './WebexAvatar';

jest.mock('../hooks/usePerson');

describe('Webex Avatar component', () => {
  test('matches snapshot with "default" status', () => {
    expect(shallow(<WebexAvatar personID="default" />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.ACTIVE}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.ACTIVE} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.BOT}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.BOT} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.CALL}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.CALL} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.DO_NOT_DISTURB}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.DO_NOT_DISTURB} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.INACTIVE}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.INACTIVE} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.MEETING}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.MEETING} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.OUT_OF_OFFICE}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.OUT_OF_OFFICE} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.PRESENTING}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.PRESENTING} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.SELF}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.SELF} />)).toMatchSnapshot();
  });

  test(`matches snapshot with "${PersonStatus.TYPING}" person status`, () => {
    expect(shallow(<WebexAvatar personID={PersonStatus.TYPING} />)).toMatchSnapshot();
  });

  test('matches snapshot with wrong personID', () => {
    expect(shallow(<WebexAvatar personID="Wrong PersonID" />)).toMatchSnapshot();
  });
});
