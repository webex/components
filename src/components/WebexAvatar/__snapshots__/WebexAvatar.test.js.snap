// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Avatar component matches snapshot with "active" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="active"
/>
`;

exports[`Webex Avatar component matches snapshot with "bot" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="bot"
/>
`;

exports[`Webex Avatar component matches snapshot with "call" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="call"
/>
`;

exports[`Webex Avatar component matches snapshot with "default" status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type={null}
/>
`;

exports[`Webex Avatar component matches snapshot with "dnd" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="dnd"
/>
`;

exports[`Webex Avatar component matches snapshot with "inactive" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="inactive"
/>
`;

exports[`Webex Avatar component matches snapshot with "meeting" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="meeting"
/>
`;

exports[`Webex Avatar component matches snapshot with "ooo" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="ooo"
/>
`;

exports[`Webex Avatar component matches snapshot with "presenting" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="presenting"
/>
`;

exports[`Webex Avatar component matches snapshot with "self" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="self"
/>
`;

exports[`Webex Avatar component matches snapshot with "typing" person status 1`] = `
<Avatar
  alt="Webex Component User"
  backgroundColor=""
  buttonClassName=""
  className={null}
  color=""
  failureBadge={false}
  hasNotification={false}
  hideDefaultTooltip={false}
  icon={null}
  isDecrypting={false}
  isOverview={false}
  onClick={null}
  size="medium"
  src="https://4b4dc97add6b1dcc891a-54bf3b4e4579920861d23cc001530c2a.ssl.cf1.rackcdn.com/V1~b33cb17c-42e3-41ac-a045-497e4002646c~b18d4572b17a4e98a16708797343ee7a~1600"
  theme={null}
  title="Webex Component User"
  type="typing"
/>
`;
